### PR TITLE
considering different current_version_keys

### PR DESCRIPTION
**Pull request by MohamedHmini**
in my case DisplayVersion doesn't exist in my regisry, but the same required data can be found under the key CurrentVersion.

### CODE REVIEW

File: src/util/winwrap.cpp [hunk 1]
```diff
@@ -602,14 +602,24 @@ namespace vind
             // include the null terminator
             auto size = static_cast<DWORD>(str.size() * sizeof(wchar_t) + 1) ;
 
-            auto result = RegGetValueW(
+            const std::wstring version_keys[2] = {L"DisplayVersion", L"CurrentVersion"};
```
Comment (pit-ray) [inline src/util/winwrap.cpp:605]: RegGetValueW takes `LPCWSTR` type, so should use one of them.

1. Type definition.
```cpp
const auto version_keys[] = 
```

2. Reference in `RegGetValueW`.
```cpp
version_key.c_str()
```

File: src/util/winwrap.cpp [hunk 2]
```diff
@@ -602,14 +602,24 @@ namespace vind
...
+            bool currentversion_found = false;
+            LSTATUS result;
+            for (std::wstring version_key : version_keys)
```
Comment (pit-ray) [inline src/util/winwrap.cpp:608]: `{` is required at the end of the for.

I recommend  `auto` reference.
```cpp
for(auto version_key : version_keys) {
````

File: src/util/winwrap.cpp [hunk 3]
```diff
@@ -602,14 +602,24 @@ namespace vind
...
-            auto result = RegGetValueW(
+            const std::wstring version_keys[2] = {L"DisplayVersion", L"CurrentVersion"};
+            bool currentversion_found = false;
+            LSTATUS result;
+            for (std::wstring version_key : version_keys) 
+                result = RegGetValueW(
                     HKEY_LOCAL_MACHINE,
                     L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion",
-                    L"DisplayVersion",
+                    version_key,
```
Comment (pit-ray) [inline src/util/winwrap.cpp:612]: `RegGetValueW` takes `LPCWSTR` type, so should refer like `version_key.c_str()`.

**Comment by pit-ray** [comment]
Thanks for the pull request. I've linked to it as a fix for the issue in #62. 

However, build seems to have failed.

This function gets a string like **21H2** to describe the environment information in the log file. It appears that the key that corresponds to `DisplayVersion` in Windows 10 1909 and earlier is `ReleaseId`.

Therefore, how about `ReleaseId` rather than `CurrentVersion`?

**Comment by MohamedHmini** [comment]
Yes ReleaseId seems right,
Sorry for not testing it before the pull request, i wasn't in a good time to prepare the env locally.

**Comment by pit-ray** [comment]
I fixed the cov-bulild check, so all checks should work.

**Comment by pit-ray** [comment]
If there is no commit within 7 days from today, I will take over this task.

**Pull request merged**